ote_sdk/ote_sdk/utils/tiler.py

```diff
@@ -7,51 +7,197 @@
+        return_features=False,
+    ):
+        """Predict on single tile
+
+        Args:
+            image (np.ndarray): full-res image
+            coord (List): tile coordinates
+            masks (List): list of raw np.ndarray masks
+            return_features (bool, optional): return saliency map and feature vector if set to true. Defaults to False.
+
+        Returns:
+            features: saliency map and feature vector
+            output: single tile prediction
+        """
+        features = [None, None]
+        offset_x, offset_y, tile_dict, tile_meta = self.preprocess_tile(image, coord)
+        raw_predictions = self.model.infer_sync(tile_dict)
+        output = self.model.postprocess(raw_predictions, tile_meta)
+        output = self.postprocess_tile(output, offset_x, offset_y, masks)
+        if return_features:
+            if "feature_vector" in raw_predictions or "saliency_map" in raw_predictions:
+                features = [
+                    raw_predictions["feature_vector"].reshape(-1),
+                    raw_predictions["saliency_map"],
                 ]
+        return features, output
+
+    def postprocess_tile(
+        self,
+        output: Union[List, Tuple],
+        offset_x: int,
+        offset_y: int,
+        masks: List,
+    ):
+        """Postprocess tile predictions
+
+        Args:
+            output (Union[List, Tuple]): predictions
+            offset_x (int): tile offset x value
+            offset_y (int): tile offset y value
+            masks (List): list of raw np.ndarray mask
+
+        Returns:
+            output: processed tile prediction
+        """
+        if self.segm:
+            tile_scores, tile_labels, tile_boxes, tile_masks = output
+            tile_boxes += np.tile([offset_x, offset_y], 2)
+            out = np.concatenate(
+                (
+                    tile_labels[:, np.newaxis],
+                    tile_scores[:, np.newaxis],
+                    tile_boxes,
+                ),
+                -1,
             )
-            offsets.append((loc_i, loc_j))  # offset y, offset x
-            if len(tiles) == self.batch_size:
-                yield tiles, offsets
-                tiles, offsets = [], []
-        if tiles:
-            yield tiles, offsets
+            masks.extend(tile_masks)
+        else:
+            assert isinstance(output, list)
+            out = detection2array(output)
+            out[:, 2:] += np.tile([offset_x, offset_y], 2)
+        return out
+
+    def preprocess_tile(self, image: np.ndarray, coord: List[int]):
+        """Preprocess Tile by cropping
+
+        Args:
+            image (np.ndarray): full-res image
+            coord (List): tile coordinates
+
+        Returns:
+            _type_: _description_
+        """
+        x1, y1, x2, y2 = coord
+        tile_dict, tile_meta = self.model.preprocess(image[y1:y2, x1:x2])
+        if self.segm:
+            tile_meta["resize_mask"] = False
```

Comment:
Model API automatically [resize masks](https://github.com/openvinotoolkit/training_extensions/blob/fadbb7c4f62b42d065bd0643c26b3798bf42247e/external/mmdetection/detection_tasks/apis/detection/model_wrappers/openvino_models.py#L72) according to their boxes during postprocess, however, in tiling phase, since we are cropping image into small patches we don't need to perform this on tile, but only on the full image.